export const catchError_ = (self, onError) => (_, sink) => {
    let innerTalkback;
    const talkback = (signal) => innerTalkback(signal);
    const replaceSource = (source, initial = false) => {
        source(0, (t, d) => {
            if (t === 0) {
                innerTalkback = d;
                if (initial) {
                    sink(0, talkback);
                }
                talkback(1);
            }
            else if (initial && t === 2 && d !== undefined) {
                replaceSource(onError(d));
            }
            else {
                sink(t, d);
            }
        });
    };
    replaceSource(self, true);
};
export const catchError = (onError) => (self) => catchError_(self, onError);
